routers/api/v1/repo/key.go

```diff
@@ -87,7 +87,7 @@ func HandleAddKeyError(ctx *context.APIContext, err error) {
 // CreateDeployKey create deploy key for a repository
 // see https://github.com/gogits/go-gogs-client/wiki/Repositories-Deploy-Keys#add-a-new-deploy-key
 func CreateDeployKey(ctx *context.APIContext, form api.CreateKeyOption) {
-	content, err := models.CheckPublicKeyString(form.Key)
+	content, err := models.CheckPublicKeyString(form.Key, ctx.Locale.Language())
```

Comment:
It should be better to not pass context to CheckPublicKeyString and hanled the error inside HandleCheckKeyStringError.